drydock_backups/patches/drydock-multipurpose-jobs

```diff
@@ -0,0 +1,127 @@
+---
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: backup-mysql-cron
+  labels:
+    app.kubernetes.io/component: cronjob
+spec:
+  suspend: false
+  schedule: {{ BACKUP_CRON_SCHEDULE }}
+  startingDeadlineSeconds: 900
+  successfulJobsHistoryLimit: 1
+  failedJobsHistoryLimit: 1
+  jobTemplate:
+    spec:
+      template:
+        spec:
+          restartPolicy: OnFailure
+          containers:
+            - name: backup
+              image: ednxops/shipyard-utils
+              env:
+                - name: MYSQL_HOST
+                  value: '{{ MYSQL_HOST }}'
+                - name: MYSQL_PORT
+                  value: '{{ MYSQL_PORT }}'
+                - name: MYSQL_ROOT_USERNAME
+                  value: '{{ MYSQL_ROOT_USERNAME }}'
+                - name: MYSQL_ROOT_PASSWORD
+                  value: '{{ MYSQL_ROOT_PASSWORD }}'
+                - name: AWS_ACCESS_KEY_ID
+                  value: '{{ BACKUPS_AWS_ACCESS_KEY }}'
+                - name: AWS_SECRET_ACCESS_KEY
+                  value: '{{ BACKUPS_AWS_SECRET_KEY }}'
+                - name: S3_BUCKET_NAME
+                  value: '{{ BACKUPS_BUCKET_NAME }}'
+                {% if BACKUP_ENDPOINT is defined %}
+                - name: BACKUP_ENDPOINT
+                  value: '{{ BACKUP_ENDPOINT }}'
+                {% endif %}
+              {% if BACKUP_K8S_USE_EPHEMERAL_VOLUMES %}
+              volumeMounts:
+                - mountPath: /data/
+                  name: backup-volume
+              {% endif %}
+              command: ["/bin/sh", "-c"]
+              args: ["FILENAME=$(date +'%Y-%m-%d').sql.gz && \
+                    mysqldump --column-statistics=0 -u $MYSQL_ROOT_USERNAME -h $MYSQL_HOST -P $MYSQL_PORT \
+                    --password=$MYSQL_ROOT_PASSWORD --all-databases --single-transaction --flush-logs \
+                    | gzip > $FILENAME && aws {% if BACKUP_ENDPOINT is defined %} --endpoint-url $BACKUP_ENDPOINT {% endif %} s3 mv $FILENAME s3://$S3_BUCKET_NAME/backups/mysql/"]
+          {% if BACKUP_K8S_USE_EPHEMERAL_VOLUMES %}
+          volumes:
+            - name: backup-volume
+              ephemeral:
+                volumeClaimTemplate:
+                  spec:
+                    accessModes:
+                      - ReadWriteOnce
+                    resources:
+                      requests:
+                        storage: {{ BACKUP_K8S_EPHEMERAL_VOLUME_SIZE }}
+          {% endif %}
+---
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: backup-mongo-cron
+  labels:
+    app.kubernetes.io/component: cronjob
+spec:
+  suspend: false
+  schedule: {{ BACKUP_CRON_SCHEDULE }}
+  startingDeadlineSeconds: 900
+  successfulJobsHistoryLimit: 1
+  failedJobsHistoryLimit: 1
+  jobTemplate:
+    spec:
+      template:
+        spec:
+          restartPolicy: OnFailure
+          containers:
+            - name: backup
+              image: ednxops/shipyard-utils
```

Comment:
Can this be a variable?

Can we also provide a Dockerfile that builds this image using this plugin?

Reply:
Now is setted as variable, thank you @MoisesGSalas .